lib/helper/Testcafe.js

```diff
@@ -0,0 +1,661 @@
+      return this._startBrowser();
+    }
+  }
+
+
+  async _before() {
+    if (this.options.restart && !this.options.manualStart) return this._startBrowser();
+    if (!this.isRunning && !this.options.manualStart) return this._startBrowser();
+  }
+
+  async _after() {
+    if (!this.isRunning) return;
+
+    if (this.options.restart) {
+      this.isRunning = false;
+      return this._stopBrowser();
+    }
+
+    if (this.options.keepBrowserState) return;
+
+    if (!this.options.keepCookies) {
+      this.debugSection('Session', 'cleaning cookies and localStorage');
+      await this.clearCookie();
+
+      // TODO IMHO that should only happen when
+      await this.executeScript(() => localStorage.clear())
+        .catch((err) => {
+          if (!(err.message.indexOf("Storage is disabled inside 'data:' URLs.") > -1)) throw err;
+        });
+    }
+  }
+
+  _afterSuite() {
+  }
+
+  async _finishTest() {
+    if (!this.options.restart && this.isRunning) return this._stopBrowser();
+  }
+
+  /**
+   * Get elements by different locator types, including strict locator
+   * Should be used in custom helpers:
+   *
+   * ```js
+   * const elements = await this.helpers['Testcafe']._locate({name: 'password'});
+   * ```
+   *
+   * {{ react }}
+   */
+  async _locate(locator) {
+    return findElements(this.context, locator);
+  }
+
+  /**
+   * {{> amOnPage }}
+   */
+  async amOnPage(url) {
+    if (!(/^\w+\:\/\//.test(url))) {
+      url = this.options.url + url;
+    }
+
+    return this.t.navigateTo(url)
+      .catch(mapError);
+  }
+
+
+  /**
+   * {{> resizeWindow }}
+   *
+   * Unlike other drivers Puppeteer changes the size of a viewport, not the window!
+   * Puppeteer does not control the window of a browser so it can't adjust its real size.
+   * It also can't maximize a window.
+   */
+  async resizeWindow(width, height) {
+    if (width === 'maximize') {
+      return this.t.maximizeWindow().catch(mapError);
+    }
+
+    return this.t.resizeWindow(width, height).catch(mapError);
+  }
+
+  /**
+   * {{> click }}
+   *
+   * {{ react }}
+   */
+  async click(locator, context = null) {
+    return proceedClick.call(this, locator, context);
+  }
+
+
+  /**
+   * {{> refreshPage }}
+   */
+  async refreshPage() {
+    // eslint-disable-next-line no-restricted-globals
+    return this.t.eval(() => location.reload(true), { boundTestRun: this.t }).catch(mapError);
+  }
+
+  /**
+   * {{> waitForVisible }}
+   *
+   * This method accepts [React selectors](https://codecept.io/react).
+   */
+  async waitForVisible(locator, sec) {
+    const timeout = sec ? sec * 1000 : undefined;
+
+    return findElements.call(this, this.context, locator)
+      .with({ visibilityCheck: true, timeout })()
+      .catch(mapError);
+  }
+
+  /**
+   * {{> fillField }}
+   * {{ react }}
+   */
+  async fillField(field, value) {
+    return this.t
+      .typeText(findElements.call(this, this.context, field), value, { replace: true })
```

Comment:
Similarly, you can get `appendField` implemented. The same but `replace: false`

Reply:
ok